share/ansible_navigator/utils/catalog_collections.py

```diff
@@ -116,62 +115,57 @@ def _generate_checksum(file_path: str, relative_path: str) -> Dict:
     def _process_plugin_dir(
         self,
         plugin_type: str,
-        filenames: List,
-        file_checksums: Dict,
-        dirpath: str,
+        filenames: Generator[Path, None, None],
+        file_checksums: Dict[str, Dict],
         collection: Dict,
     ) -> None:
-        # pylint: disable=too-many-arguments
         """Process each plugin within one plugin directory.
 
         :param plugin_type: The type of plugins
         :param filenames: The filenames of the plugins
         :param file_checksums: The checksums for the plugin files
-        :param dirpath: The path of the directory containing the plugins
         :param collection: The details of the collection
         """
         for filename in filenames:
-            file_path = f"{dirpath}/{filename}"
-            relative_path = file_path.replace(collection["path"], "")
-            _basename, extension = os.path.splitext(filename)
-            if not filename.startswith("__") and extension == ".py":
-                checksum_dict = file_checksums.get(relative_path)
+            relative_path = Path(filename).relative_to(collection["path"])
+            if not str(filename).startswith("__"):
+                checksum_dict = file_checksums.get(str(relative_path))
                 if not checksum_dict:
-                    checksum_dict = self._generate_checksum(file_path, relative_path)
+                    checksum_dict = self._generate_checksum(filename, relative_path)
                 checksum = checksum_dict[f"chksum_{checksum_dict['chksum_type']}"]
                 collection["plugin_checksums"][checksum] = {
-                    "path": relative_path,
+                    "path": str(relative_path),
                     "type": plugin_type,
                 }
 
-    def _one_path(self, directory: str) -> None:
+    def _one_path(self, directory: Path) -> None:
         """Process the contents of an <...>/ansible_collections/ directory.
 
         :param directory: The path to collections directory to walk and load
         """
-        for directory_path in glob(f"{directory}/*/*/"):
-            manifest_file = f"{directory_path}/MANIFEST.json"
-            galaxy_file = f"{directory_path}/galaxy.yml"
+        for directory_path in directory.glob("*/*/"):
+            manifest_file = Path(directory_path, "MANIFEST.json")
+            galaxy_file = Path(directory_path, "galaxy.yml")
```

Comment:
`directory_path` is already a `Path` object, no need to reconstruct it:
```suggestion
            manifest_file = directory_path / "MANIFEST.json"
            galaxy_file = directory_path / "galaxy.yml"
```

Reply:
Cool, although odd.

```
>>> directory_path = pathlib.Path("/home/bthornto")
>>> directory_path.__truediv__("file.txt")
PosixPath('/home/bthornto/file.txt')
```

Reply:
Well, that's one of the main selling points of the pathlib interface. It's supposed to be the canonical way of using it.